packages/libs/contracts-sdk/contracts/facets/VincentAppFacet.sol

```diff
@@ -164,6 +164,50 @@ contract VincentAppFacet is VincentBase {
         emit LibVincentAppFacet.DelegateeRemoved(appId, delegatee);
     }
 
+    /**
+     * @notice Replace all delegatees for an app with the provided array. Allows adding empty arrays effectively replacing all delegatees at once.
+     * @dev Only the app manager can set delegatees. This function removes all existing delegatees
+     *      and adds the new ones. Each delegatee can only be associated with one app at a time.
+     * @param appId ID of the app
+     * @param delegatees Array of addresses to set as delegatees
+     */
+    function setDelegatee(uint256 appId, address[] calldata delegatees)
+        external
+        appNotDeleted(appId)
+        onlyAppManager(appId)
+        onlyRegisteredApp(appId)
+    {
+        VincentAppStorage.AppStorage storage as_ = VincentAppStorage.appStorage();
+        VincentAppStorage.App storage app = as_.appIdToApp[appId];
+
+        address[] memory currentDelegatees = app.delegatees.values();
+        
+        // Remove all current delegatees
+        for (uint256 i = 0; i < currentDelegatees.length; i++) {
+            app.delegatees.remove(currentDelegatees[i]);
+            as_.delegateeAddressToAppId[currentDelegatees[i]] = 0;
+            emit LibVincentAppFacet.DelegateeRemoved(appId, currentDelegatees[i]);
+        }
```

Comment:
I think storing `currentDelegatees[i]` in a local var as done in the next for loop offers some gas savings